src/components/OrderSummaryPlanSelect.js

```diff
@@ -1,18 +1,62 @@
+import { useState } from 'react';
+import DurationMenu from './DurationMenu';
+
 import icon from '../images/icon-music.svg';
 
 function OrderSummaryPlanSelect () {
 
+    // Tempary test data to test functionality on this component and DurationMenu
+    const planOptions = [
+        {
+            duration: 'Annual',
+            price: 59.99
+        },
+        {
+            duration: 'Monthly',
+            price: 6
+        },
+    ];
+    const [displayDurationMenu, setDisplayDurationMenu] = useState(false);
+    const [selectedOption, setSelectedOption] = useState(0);
+
+    // Switches from displaying the selected option to displaying the DurationMenu
+    const onClickHandler = (e) => {
+        e.preventDefault();
+        toggleDurationMenu();
+    }
+
+    // Handler takes 'duration' property passed up from DurationMenu and makes it the selected option
+    const onSelectHandler = (selected) => {
+        toggleDurationMenu();
+        setSelectedOption(selected === 'Annual' ? 0 : 1);
+    }
+
+    // Toggle function shared by handlers
+    const toggleDurationMenu = () => {
+        setDisplayDurationMenu(!displayDurationMenu);
+    }
+
+    const durationToTimeSpan = (duration) => {
+        return duration === 'Annual' ? 'year' : 'month';
+    }
+
     return (
-        <div className='my-4 py-4 bg-very-pale-blue rounded-xl flex justify-around items-center' id='plan-selector'>
-            <div className='flex items-center'>
-                <img className='pr-4' src={icon} alt='' />
-                <div className=''>
-                    <h3 className='font-bold p-1'>Annual Plan</h3>
-                    <p className='text-desaturated-blue'>$59.99/year</p>
+        <>  
+            {displayDurationMenu && <DurationMenu options={planOptions} onSelect={onSelectHandler} /> }  
+
+            {!displayDurationMenu && 
+                <div className='my-4 py-4 bg-very-pale-blue rounded-xl flex justify-around items-center' id='plan-selector'>
+                    <div className='flex items-center'>
+                        <img className='pr-4' src={icon} alt='' />
+                        <div className=''>
+                            <h3 className='font-bold p-1'>{planOptions[selectedOption].duration} Plan</h3>
+                            <p className='text-desaturated-blue'>${`${planOptions[selectedOption].price}/${durationToTimeSpan(planOptions[selectedOption].duration)}`}</p>
```

Comment:
Is the outer set of `{}` needed here?

Reply:
Yes, because I am adding JavaScript to the JSX. Though I've not formatted this template literal the same way as a similar one in DurationMenu.js (this one has the dollar sign outside of the {}) so I should make that consistent.